packages/api/endpoints/granules.js

```diff
@@ -274,19 +275,26 @@ async function del(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function get(req, res) {
+  const {
+    granulePgModel = new GranulePgModel(),
+    knex = await getKnexClient(),
+  } = req.testContext || {};
   const { getRecoveryStatus } = req.query;
   const granuleId = req.params.granuleName;
-  let result;
+  let granule;
   try {
-    result = await (new Granule()).get({ granuleId });
+    granule = await granulePgModel.get(knex, { granule_id: granuleId });
   } catch (error) {
-    if (error.message.startsWith('No record found')) {
+    if (error instanceof RecordDoesNotExist) {
       return res.boom.notFound('Granule not found');
     }
 
     throw error;
   }
 
+  // Get related files, execution ARNs, provider, PDR, and collection and format
+  const result = await translatePostgresGranuleToApiGranule(granule, knex);
```

Comment:
This method does a lot. I think it makes sense to keep the files, executions, provider, PDR, and collection associations in the translate function but there are a few different ways to split this up if that seems like too much.

Reply:
I think it's probably fine to have this in the granules object translator, despite pulling in other entities given our API granule structure.  That's consistent with other translators, although they're only using the models to get IDs and the like. 

Is the alternate approach you're thiking of something more along the lines of keeping the translator 'pure' in that it only brings back an object with reference values and then there's more endpoint logic that gets each other record and translates/assembles it?  

something like (or the curried version) of something like await getExternalGranuleRecordIdentifiers(getApiGraunleFiels(translatePostgresGranuleToApiGranule(granule,knex)))?  

The tests might be better, and the abstraction might dry up some of our translators, but I think this is acceptable if the units cover the logic adaquately.